d2game/d2player/game_controls.go

```diff
@@ -444,6 +444,12 @@ func (g *GameControls) onEscKey() {
 	}
 
 	if g.inventory.IsOpen() {
+		if g.inventory.moveGoldPanel.IsOpen() {
+			g.inventory.moveGoldPanel.Close()
+
+			return
+		}
+
 		g.inventory.Close()
 
 		escHandled = true
```

Comment:
This section of the code does not scale well at all, i think it needs a refactor. I'm thinking that as panels are opened/closed they should be pushed/popped from a stack `[]interface{}`. This could be stored inside of `GameControls` as `GameControls.panelStack` or something similar.

```golang
type panel interface {
	IsOpen() bool
	Close()
}

type panelContainer interface {
	HasOpenPanels() bool
	ClosePanels()
}

func (g *GameControls) onEscKey() {
	// special case for opening the escape menu when panel stack is empty
	if len(g.panelStack) == 0 {
		g.escapeMenu.Open()

		g.panelStack = append(g.panelStack, g.escapeMenu)

		g.updateLayout()

		return
	}

	// generic case, pop the last thing off the, close it, and then update
	for idx := range g.panelStack {
		escHandled := false
		
		if p, isPanel := g.panelStack[idx].(panel); isPanel {
			p.Close()
			escHandled = true
		}

		if p, isPanelContainer := g.panelStack[idx].(panelContainer); isPanelContainer {
			p.ClosePanels()
			escHandled = true
		}

		if escHandled {
			g.panelStack = append(g.panelStack[:idx], g.panelStack[idx+1:]...)
			g.updateLayout()
			break
		}
	}
}
```

However, this would require that every panel to these interfaces (`helpOverlay.Toggle` is used in the current code...)

Reply:
I've opened an issue (#965) for refactoring this part of the code, I think that this PR is good to go.